### PR TITLE
disable webhook enforcer on azure

### PR DESCRIPTION
**Pull request by luciferinlove**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets |
| License         | Apache 2.0

### What's in this PR?
Disable admission webhook enforcer on Azure

### Why?
This caused an eternal operator reconciliation.

### Checklist
- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: internal/assets/manifests/istio-discovery/templates/mutatingwebhook.yaml [hunk 1]
```diff
@@ -54,6 +54,8 @@ metadata:
     istio.io/rev: {{ include "namespaced-revision" . }}
     app: sidecar-injector
     release: {{ .Release.Name }}
+  annotations:
+    admissions.enforcer/disabled: "true"
```
Comment (vhosakot) [inline internal/assets/manifests/istio-discovery/templates/mutatingwebhook.yaml:58]: adding link to docs for reference: https://learn.microsoft.com/en-us/azure/aks/faq#can-admission-controller-webhooks-impact-kube-system-and-internal-aks-namespaces

**Review by vhosakot — APPROVED** [review]
lgtm

**Review by tiswanso — APPROVED** [review]
lgtm... the Azure admission enforcer wouldn't have any effect anyway since istio-operator already uses namespace selectors for istio-revs.

**Review by shanchunyang0919 — APPROVED** [review]
Looks good to me. 

Nitpick: Would you mind adding some info to the docs explaining why we are adding the annotation  `admissions.enforcer/disabled:` in the helm charts? Thanks.

**Review by nishantapatil3 — APPROVED** [review]
Thanks @LuciferInLove. LGTM!

**Pull request merged**